### PR TITLE
61 js refactor services

### PR DESCRIPTION
**Pull request by berggren**
# First round of JS refactoring - Services

This is the first round on refactoring the Angular JS code. See issue #46 for tracking of the different steps.

Move all code that talks to the Timesketch REST API to services. Test will be added at a later time but with this refactor it will at least be possible. :)

Ref issue: #61 
Reviewer: @mbushkov

### CODE REVIEW

File: timesketch/ui/static/js/controllers.js [hunk 1]
```diff
@@ -104,21 +98,17 @@ timesketch.controller('TsExploreCtrl', function($scope, $http) {
...
-                var id = data.objects[0].id;
-                var view_url = '/sketch/' + $scope.sketch_id + '/explore/view/' + id + '/';
-                window.location.href = view_url;
+            timesketchApi.saveView(
+                $scope.sketch_id, $scope.view_name, $scope.query, $scope.filter)
+                .success(function(data) {
+                    var id = data.objects[0].id;
+                    var view_url = '/sketch/' + $scope.sketch_id + '/explore/view/' + id + '/';
+                    window.location.href = view_url;
             });
```
Comment (mbushkov) [inline timesketch/ui/static/js/controllers.js]: It probably makes sense to handle requests failures in some generic way. For example, timesketchApi may expose a variable with the last error and then you can a piece of UI watching on it. Then, when error happens - no matter when or during which API call - you'll see it in the UI. This way you'll also won't have to manually handle failures everywhere.

**Comment by mbushkov** [comment]
LGTM! Looks really good, and sorry for taking so long to review.

File: timesketch/ui/static/js/controllers.js (hunk 1 above)
Comment (berggren) [inline timesketch/ui/static/js/controllers.js]: Great idea! I added a interceptor for response failures that broadcasts http-error. This is picked up by a directive that shows a butterbar telling something went wrong.

I will revisit this in the future to make this more elegant.

File: timesketch/ui/static/js/controllers.js [hunk 2]
```diff
@@ -104,21 +102,17 @@ timesketch.controller('TsExploreCtrl', function($scope, $http) {
...
-            $http.post('/api/v1/sketches/' + $scope.sketch_id + '/views/', params).success(function(data) {
-                var id = data.objects[0].id;
-                var view_url = '/sketch/' + $scope.sketch_id + '/explore/view/' + id + '/';
-                window.location.href = view_url;
+            timesketchApi.saveView(
+                $scope.sketch_id, $scope.view_name, $scope.query, $scope.filter)
+                .success(function(data) {
+                    var id = data.objects[0].id;
+                    var view_url = '/sketch/' + $scope.sketch_id + '/explore/view/' + id + '/';
+                    window.location.href = view_url;
```
Comment (mbushkov) [inline timesketch/ui/static/js/controllers.js]: Nit: you can use $window Angular alias: https://docs.angularjs.org/api/ng/service/$window
Then you'll have to inject it via dependency injection mechanism and will be able to mock it out in tests.

**Pull request merged**